lms/djangoapps/courseware/access_response.py

```diff
@@ -21,13 +22,16 @@ def __init__(self, has_access, error_code=None, developer_message=None, user_mes
                 to show the developer
             user_message (String): optional - default is None. Message to
                 show the user
+            detailed_user_message (String): optional - default is None. Message to
+                show the user when additional context like the course name is necessary
```

Comment:
I think it might be worth being really clear about what this `detailed_user_message` should be used (and where `user_message`/`user_fragment` should be used instead). In particular, the usecase that you added it for was for putting it somewhere where the context of the course wasn't implied. But that context seems specific to only some error messages. Would `no_context_user_message` make it clearer? Or maybe it should just talk in the docstring about how this is for error messages that appear without any surrounding context? Or maybe this is `banner_user_message` (because the banner is where most out-of-context messages happen)?

I'm not sure, but given the number of types of messages this object is growing, it seems worth trying to clarify a bit.

Reply:
I'm on board with making the use cases for this message clearer.  
I'm also concerned about being too specific, since that may cause to need more fields than necessary.  
For example, if we call it banner_user_message, we might need to provide different context on different banners for some reason. no_context_user_message signals to me that we aren't providing additional context. How about I call it additional_context_user_message and mention in the docstring that this is intended to be used in situations where the user message doesn't provide enough context, such as when we need to mention which course we are referencing. (And that the fragment is intended for html)